cpu/native/netdev_raw802154/netdev_raw802154.c

```diff
@@ -0,0 +1,337 @@
+    (void) fd;
+    (void) arg;
+    netdev_t *netdev = (netdev_t *)&netdev_raw802154;
+
+    if (netdev->event_callback)
+        netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+}
+
+static void _set_addr_short(netdev_raw802154_t *dev, uint16_t addr)
+{
+    dev->netdev.short_addr[0] = (uint8_t)(addr);
+    dev->netdev.short_addr[1] = (uint8_t)(addr >> 8);
+}
+
+static void _set_addr_long(netdev_raw802154_t *dev, uint64_t addr)
+{
+    for (int i = 0; i < 8; i++)
+        dev->netdev.long_addr[i] = (uint8_t)(addr >> (i * 8));
+}
+
+static void _set_pan(netdev_raw802154_t *dev, uint16_t pan)
+{
+    dev->netdev.pan = pan;
+}
+
+static int16_t _get_txpower(netdev_raw802154_t *dev)
+{
+    (void)dev;
+    return 0;
+}
+
+static uint8_t _get_max_retries(netdev_raw802154_t *dev)
+{
+    (void)dev;
+    return 0;
+}
+
+#define _MAX_MHR_OVERHEAD   (25)
+static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+
+    switch (opt) {
+        case NETOPT_STATE:
+            if (max_len < sizeof(netopt_state_t)) {
+                return -EOVERFLOW;
+            }
+            *((netopt_state_t *)val) = NETOPT_STATE_RX;
+            return sizeof(netopt_state_t);
+        case NETOPT_CHANNEL_PAGE:
+            if (max_len < sizeof(int16_t)) {
+                return -EOVERFLOW;
+            }
+            ((uint8_t *)val)[1] = 0;
+            ((uint8_t *)val)[0] = _get_page(dev);
+            return sizeof(uint16_t);
+        case NETOPT_MAX_PACKET_SIZE:
+            if (max_len < sizeof(int16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = 127 - _MAX_MHR_OVERHEAD;
+            return sizeof(uint16_t);
+        case NETOPT_TX_POWER:
+            if (max_len < sizeof(int16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = _get_txpower(dev);
+            return sizeof(uint16_t);
+            break;
+        case NETOPT_CSMA:
+            if (max_len < sizeof(netopt_enable_t)) {
+                return -EOVERFLOW;
+            }
+            *((netopt_enable_t *)val) = NETOPT_DISABLE;
+            return sizeof(netopt_enable_t);
+        case NETOPT_AUTOCCA:
+            if (max_len < sizeof(netopt_enable_t)) {
+                return -EOVERFLOW;
+            }
+            *((netopt_enable_t *)val) = NETOPT_DISABLE;
+            return sizeof(netopt_enable_t);
+        case NETOPT_RETRANS:
+            if (max_len < sizeof(uint8_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint8_t *)val) = _get_max_retries(dev);
+            return sizeof(uint8_t);
+            break;
+        default:
+            break;
+    }
+    int res;
+    if (((res = netdev_ieee802154_get((netdev_ieee802154_t *)netdev, opt, val,
+                                       max_len)) >= 0) || (res != -ENOTSUP)) {
+        return res;
+    }
+    return 0;
```

Comment:
This function should return `-ENOTSUP` if the option is not supported. During testing some options were called and consequently printed (namely `NETOPT_BANDWIDTH` (`BW` in `ifconfig`), `NETOPT_SPREADING_FACTOR` (`SF`) and `NETOPT_CODING_RATE` (`CR`), that your device did not support, so they generate broken (including sometimes non-ASCII character containing => Linux shell output breaking) output:

```
> ifconfig
Iface  7  HWaddr: fe:ff  Channel: 11  Frequency: 0Hz  Page: 0  NID: 0x23 BW: `V`V`V`V`V`VkHz  SF: 247  CR: `V`V`V  Link: up 
          Long HWaddr: 06:2d:1c:4f:6b:49:66:f7 
           TX-Power: 0dBm  State: RX  max. Retrans.: 0 
          MTU:1280  HL:64  RTR  
          IPHC  
          Source address length: 8
          Link type: wireless
          inet6 addr: fe80::42d:1c4f:6b49:66f7  scope: local  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff49:66f7
          inet6 group: ff02::1a
          
          Statistics for Layer 2
            RX packets 0  bytes 0
            TX packets 3 (Multicast: 3)  bytes 115
            TX succeeded 3 errors 0
          Statistics for IPv6
            RX packets 0  bytes 0
            TX packets 3 (Multicast: 3)  bytes 178
            TX succeeded 3 errors 0
```

Reply:
This has nothing to do with `phy`, just with the return value of this function. It should be `-ENOTSUP`, if the option is not provided. Not 0.

Reply:
pretty sure I copied that from somewhere else. :-)

Reply:
Welp, that doesn't make it correct ;-P. But I also see right now that the [doc](http://doc.riot-os.org/structnetdev__driver.html#a0f7bbd084d7a51756fba665903884317) is outright wrong on that (will fix). If correct, the driver is supposed to return >0 (actual length of value) on success, <0 on error (`-ENOTSUP` when not supported). Would be great to know where you copy-pasta'd that from so we can fix it there as well ;-).